go/tests/googleai_test.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"cmp"
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"slices"
+	"testing"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/firebase/genkit/go/core"
+	"github.com/firebase/genkit/go/core/tracing"
+	"github.com/firebase/genkit/go/internal"
+	"github.com/firebase/genkit/go/plugins/googleai"
+	gocmp "github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/jba/xltest/go/xltest"
+	"google.golang.org/api/option"
+)
+
+var update = flag.Bool("update", false, "update golden files with results")
+
+func TestGoogleAI(t *testing.T) {
+	ctx := context.Background()
+	err := googleai.Init(ctx, googleai.Config{
+		ClientOptions: []option.ClientOption{option.WithHTTPClient(mockClient())},
+		Models:        []string{"gemini-1.5-pro"},
+		Embedders:     []string{"embedding-001"},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("embed", func(t *testing.T) {
+		test, err := xltest.ReadFile(filepath.Join("testdata", "googleai-embedder.yaml"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		valfunc := validateEmbedder
+		if *update {
+			valfunc = updateEmbedder
+		}
+		test.Run(t, func(text string) (embedderResult, error) {
```

Comment:
This will be clearer if we pull the function literal into its own variable, so that the `valfunc` variable isn't hidden.

Reply:
done